### PR TITLE
Macro parser improvements

### PR DESCRIPTION
**Pull request by sdroege**
Backport of https://github.com/gtk-rs/gtk4-rs/pull/103

### CODE REVIEW

File: gtk3-macros/src/attribute_parser.rs [hunk 1]
```diff
@@ -0,0 +1,180 @@
+use proc_macro2::Span;
```
Comment (GuillaumeGomez) [inline gtk3-macros/src/attribute_parser.rs:1]: No header in this crate?

Comment (sdroege) [reply gtk3-macros/src/attribute_parser.rs:1]: That's how it apparently always was. So we need to fix that after this PR, and also in gtk4 (CC @bilelmoussaoui )

Comment (GuillaumeGomez) [reply gtk3-macros/src/attribute_parser.rs:1]: Ok, I'll let you open the issue then. ;)

**Comment by GuillaumeGomez** [comment]
Thanks!

**Comment by sdroege** [comment]
@GuillaumeGomez You create an issue?

File: gtk3-macros/src/attribute_parser.rs (hunk 1 above)
Comment (sdroege) [reply gtk3-macros/src/attribute_parser.rs:1]: Ah you said I should, ok!

**Pull request merged**